packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Statements/AttributeStatement.cs

```diff
@@ -12,30 +11,63 @@ public sealed class AttributeStatement : MethodBodyStatement
     {
         public CSharpType Type { get; }
         public IReadOnlyList<ValueExpression> Arguments { get; }
+        public IReadOnlyList<KeyValuePair<string, ValueExpression>> PositionalArguments { get; }
 
-        public AttributeStatement(CSharpType type, IReadOnlyList<ValueExpression> arguments)
+        public AttributeStatement(CSharpType type, IReadOnlyList<ValueExpression> arguments, IReadOnlyList<KeyValuePair<string, ValueExpression>> positionalArguments)
         {
             Type = type;
             Arguments = arguments;
+            PositionalArguments = positionalArguments;
         }
 
-        public AttributeStatement(CSharpType type, params ValueExpression[] arguments) : this(type, (IReadOnlyList<ValueExpression>)arguments) { }
+        public AttributeStatement(CSharpType type, IReadOnlyList<ValueExpression> arguments) : this(type, arguments, []) { }
+
+        public AttributeStatement(CSharpType type, IReadOnlyList<KeyValuePair<string, ValueExpression>> positionalArguments) : this(type, [], positionalArguments) { }
+
+        public AttributeStatement(CSharpType type, params ValueExpression[] arguments) : this(type, arguments, []) { }
 
         internal override void Write(CodeWriter writer)
         {
-            if (Arguments.Any())
+            writer.Append($"[{Type}");
+            var hasArguments = Arguments.Count > 0 || PositionalArguments.Count > 0;
+            if (hasArguments)
+            {
+                writer.AppendRaw("(");
+            }
+            if (Arguments.Count > 0)
+            {
+                for (int i = 0; i < Arguments.Count; i++)
+                {
+                    Arguments[i].Write(writer);
+                    if (i != Arguments.Count - 1)
+                    {
+                        writer.AppendRaw(", ");
+                    }
+                }
+            }
+            if (PositionalArguments.Count > 0)
             {
-                writer.Append($"[{Type}(");
-                foreach (var argument in Arguments)
+                if (Arguments.Count > 0)
+                {
+                    writer.AppendRaw(", ");
+                }
+                int i = 0;
+                foreach (var (key, value) in PositionalArguments)
```

Comment:
nit: might be easier to use regular for loop to avoid incrementing ourselves.

Reply:
sure, this is actually the code for dictionary, since we change it to a list, we should change it